packages/network-enablement-controller/src/NetworkEnablementController.ts

```diff
@@ -0,0 +1,353 @@
+    });
+
+    messenger.subscribe('NetworkController:networkRemoved', ({ chainId }) => {
+      this.#removeNetworkEntry(chainId);
+    });
+  }
+
+  setEnabledNetwork(chainId: Hex | CaipChainId): void {
+    this.#toggleNetwork(chainId, true);
+  }
+
+  setDisabledNetwork(chainId: Hex | CaipChainId): void {
+    this.#toggleNetwork(chainId, false);
+  }
+
+  isNetworkEnabled(chainId: Hex | CaipChainId): boolean {
+    try {
+      const { namespace, storageKey } = this.#deriveKeys(chainId);
+      return Boolean(this.state.enabledNetworkMap[namespace]?.[storageKey]);
+    } catch {
+      return false;
+    }
+  }
+
+  getEnabledNetworksForNamespace(namespace: CaipNamespace): string[] {
+    return Object.entries(this.state.enabledNetworkMap[namespace] ?? {})
+      .filter(([, enabled]) => enabled)
+      .map(([id]) => id);
+  }
+
+  getAllEnabledNetworks(): Record<CaipNamespace, string[]> {
+    return (
+      Object.keys(this.state.enabledNetworkMap) as CaipNamespace[]
+    ).reduce(
+      (acc, ns) => {
+        acc[ns] = this.getEnabledNetworksForNamespace(ns);
+        return acc;
+      },
+      {} as Record<CaipNamespace, string[]>,
+    );
+  }
+
+  // ---------------------------- Internals ----------------------------------
+
+  #deriveKeys(chainId: Hex | CaipChainId) {
+    const caipId: CaipChainId = isHexString(chainId)
+      ? toEvmCaipChainId(chainId as Hex)
+      : (chainId as CaipChainId);
+    const { namespace, reference } = parseCaipChainId(caipId);
+    let storageKey: string;
+    if (namespace === (KnownCaipNamespace.Eip155 as string)) {
+      storageKey = isHexString(chainId)
+        ? (chainId as string)
+        : toHex(reference);
+    } else {
+      storageKey = caipId;
+    }
+    return { namespace, storageKey, caipId } as const;
+  }
+
+  #ensureNamespaceBucket(
+    state: NetworkEnablementControllerState,
+    ns: CaipNamespace,
+  ) {
+    if (!state.enabledNetworkMap[ns]) {
+      state.enabledNetworkMap[ns] = {} as Record<string, boolean>;
+    }
+  }
+
+  #ensureNetworkEntry(chainId: Hex | CaipChainId, enable = false): void {
+    const { namespace, storageKey } = this.#deriveKeys(chainId);
+    this.update((s) => {
+      this.#ensureNamespaceBucket(s, namespace);
+      if (!(storageKey in s.enabledNetworkMap[namespace])) {
+        s.enabledNetworkMap[namespace][storageKey] = enable;
+      }
+    });
+  }
+
+  #removeNetworkEntry(chainId: Hex | CaipChainId): void {
+    const { namespace, storageKey } = this.#deriveKeys(chainId);
+    this.update((s) => {
+      delete s.enabledNetworkMap[namespace]?.[storageKey];
+      if (Object.keys(s.enabledNetworkMap[namespace] ?? {}).length === 0) {
+        delete s.enabledNetworkMap[namespace];
+      }
+      this.#ensureAtLeastOneEnabled(s);
+    });
+  }
+
+  #ensureAtLeastOneEnabled(state: NetworkEnablementControllerState) {
+    const anyEnabled = Object.values(state.enabledNetworkMap).some((map) =>
+      Object.values(map).some(Boolean),
```

Comment:
Nit: The values are already boolean, so what are your thoughts on being explicit here?

``` suggestion
      Object.values(map).some((isEnabled) => isEnabled),